pyro/contrib/epidemiology/sir.py

```diff
@@ -115,3 +94,115 @@ def transition_bwd(self, params, prev, curr, t):
+    recovered individuals (the recovered individuals are implicit: ``R =
+    population - S - I``) with transitions ``S -> I -> R``.
+
+    This model accounts for superspreading (overdispersed individual
+    reproductive number) by assuming each infected individual infects
+    BetaBinomial-many susceptible individuals, where the BetaBinomial
+    distribution acts as an overdispersed Binomial distribution, adapting the
+    more standard NegativeBinomial distribution that acts as an overdispersed
+    Poisson distribution [1,2] to the setting of finite populations. To
+    preserve Markov structure, we follow [2] and assume all infections by a
+    single individual occur on the single time step where that individual makes
+    an ``I -> R`` transition.
+
+    **References**
+
+    [1] J. O. Lloyd-Smith, S. J. Schreiber, P. E. Kopp, W. M. Getz (2005)
+        "Superspreading and the effect of individual variation on disease
+        emergence"
+        https://www.nature.com/articles/nature04153.pdf
+    [2] Lucy M. Li, Nicholas C. Grassly, Christophe Fraser (2017)
+        "Quantifying Transmission Heterogeneity Using Both Pathogen Phylogenies
+        and Incidence Time Series"
+        https://academic.oup.com/mbe/article/34/11/2982/3952784
+
+    :param int population: Total ``population = S + I + R``.
+    :param float recovery_time: Mean recovery time (duration in state
+        ``I``). Must be greater than 1.
+    :param iterable data: Time series of new observed infections.
+    :param int data: Time series of new observed infections, i.e. a Binomial
+        subset of the ``S -> I`` transitions at each time step.
+    """
+
+    def __init__(self, population, recovery_time, data):
+        compartments = ("S", "I")  # R is implicit.
+        duration = len(data)
+        super().__init__(compartments, duration, population)
+
+        assert isinstance(recovery_time, float)
+        assert recovery_time > 1
+        self.recovery_time = recovery_time
+
+        self.data = data
+
+    series = ("S2I", "I2R", "obs")
+    full_mass = [("R0", "rho", "k")]
+
+    def global_model(self):
+        tau = self.recovery_time
+        R0 = pyro.sample("R0", dist.LogNormal(0., 1.))
+        k = pyro.sample("k", dist.Exponential(1.))
+        rho = pyro.sample("rho", dist.Uniform(0, 1))
+
+        # Convert interpretable parameters to distribution parameters.
+        prob_i = 1 / tau
+
+        return R0, k, prob_i, rho
+
+    def initialize(self, params):
+        # Start with a single infection.
+        return {"S": self.population - 1, "I": 1}
+
+    def transition_fwd(self, params, state, t):
+        R0, k, prob_i, rho = params
+
+        # Sample flows between compartments.
+        I2R = pyro.sample("I2R_{}".format(t),
+                          dist.Binomial(state["I"], prob_i))
+        c1 = (k * I2R).clamp(min=1e-6)
+        c0 = k / R0 * self.population
+        S2I = pyro.sample("S2I_{}".format(t),
```

Comment:
why not use the dist helper?

Reply:
oops, I must have missed this. Will fix!